man/kernel-install.xml

```diff
@@ -224,7 +223,7 @@
       (EFI System Partition) are mounted, and also conceptually referred to as <varname>$BOOT</varname>. Can
       be overridden by setting <varname>$BOOT_ROOT</varname> (see below).</para>
 
-      <para><varname>$KERNEL_INSTALL_LAYOUT=bls|uki|other|...</varname> is set for the plugins to specify the
```

Comment:
So I think we should explicitly support a mode where type1 entries are generated for UKIs for people who want to set kernel cmdline for example. In fact in sd-boot we should explicitly support that if a type1 and a type2 entry we find share the same name we'll ignore the type2 entry, so that people can "wrap" a type2 kernel as type1 if they like, to paramaterize it further.

(of course, this will come with implications on secureboot, i.e. what will actually be allowed, but that's another question).

hence, I think three scenarios matter in the long run

1. classic non-UKI kernel package, with local initrd, installed as type1
1. UKI packaged, and installed 1:1 as type2
2. UKI packaged, and installed with a type1 maybe because /etc/kernel/cmdline was set or some other reason (maybe a devicetree file or so)
3. classic non-UKI kernel packaged, from which a local UKI is generated/signed, and installed as type1

hence maybe:

KERNEL_INSTALL_LAYOUT=bls|uki|auto|other

bls: always install as type 1. i.e. for packaged classic non-UKI kernels, install as regular type1. and for UKI kernels generate a type2 wrapper.
uki: always install as type 2. i.e. for packaged classic non-UKI kernels, generate UKI and install as UKI. for UKI kernels install 1:1
auto: try to be smart, i.e. install classic kernels as type1, and UKIs as type2. Except: if a local signing key is found in /etc/kernel/uki-sign.pem (or so), then take that as hint to generate a UKI from classic kenrels, and sign them (i.e. if people drop-in stuff upgrade classic kernels to UKI mode). And the opposite direction too: if /etc/kernel/cmdline exists, install type1 wrapper for uki with it included.

but I'd see that as the long term perspective. we should keep this in mind even if we don't implement all this right-away.